packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -41,6 +41,51 @@ class RenderableTiledMap {
       ? batchesByLayer.first
       : <String, SpriteBatch>{};
 
+  /// Changes the visibility of the corresponding layer, if different
+  void setLayerVisibility(int layerId, bool visibility) {
+    if (map.layers[layerId].visible != visibility) {
+      map.layers[layerId].visible = visibility;
+      refreshCache();
+    }
+  }
+
+  /// Gets the visibility of the corresponding layer
+  bool getLayerVisibility(int layerId) {
+    return map.layers[layerId].visible;
+  }
+
+  /// Changes the Gid of the corresponding layer at the given position,
+  /// if different
+  void setTileData({
+    required int layerId,
+    required int x,
+    required int y,
+    required Gid gid,
+  }) {
+    final layer = map.layers[layerId] as TileLayer;
+    final td = layer.tileData;
+    if (td != null) {
+      if (td[y][x].tile != gid.tile ||
+          td[y][x].flips.horizontally != gid.flips.horizontally ||
+          td[y][x].flips.vertically != gid.flips.vertically ||
+          td[y][x].flips.diagonally != gid.flips.diagonally) {
+        td[y][x] = gid;
+        refreshCache();
+      }
+    }
+  }
+
+  /// Gets the Gid  of the corresponding layer at the given position
+  Gid? getTileData({required int layerId, required int x, required int y}) {
+    final layer = map.layers[layerId] as TileLayer;
```

Comment:
How do you know that this particular `Layer` is a `TileLayer`?

Reply:
Better now?